app/backend/src/couchers/servicers/auth.py

```diff
@@ -154,11 +154,20 @@ def SignupFlow(self, request, context):
                     existing_user = session.execute(
                         select(User).where(User.email == request.basic.email)
                     ).scalar_one_or_none()
+                   
+
                     if existing_user:
                         context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.SIGNUP_FLOW_EMAIL_TAKEN)
                     existing_flow = session.execute(
                         select(SignupFlow).where(SignupFlow.email == request.basic.email)
                     ).scalar_one_or_none()
```

Comment:
You've put the existing username check between the existing flow logic.

Move what you have to be above this `existing_flow = ...` statement

Reply:
i have moved above the existing flow logic